include/zcbor_common.h

```diff
@@ -115,12 +115,14 @@ union {
 	uint8_t const *payload_end; /**< The end of the payload. This will be
 	                                 checked against payload before
 	                                 processing each element. */
-	bool indefinite_length_array; /**< Is set to true if the decoder is currently
-	                                   decoding the contents of an indefinite-
-	                                   length array. */
 	bool payload_moved; /**< Is set to true while the state is stored as a backup
 	                         if @ref zcbor_update_state is called, since that function
 	                         updates the payload_end of all backed-up states. */
+struct {
+	bool indefinite_length_array; /**< Is set to true if the decoder is currently
+	                                   decoding the contents of an indefinite-
+	                                   length array. */
+} d; /* Decode state */
```

Comment:
The parent struct `zcbor_state_t` is used for both encoding and decoding. This `d` struct encapsulates the portion that is only used when decoding. `d` is because of "decode" and to keep it short when being used. The point of having a named struct like this is to be able to bulk copy that part of the zcbor_state_t when using `ZCBOR_FLAG_KEEP_DECODE_STATE`. Sounds like this stuff is missing some documentation. Where do you think it's best to have it? Here in the header file, or maybe in the README?

Reply:
I would argue that calling it `decode_state` and spending some more characters in the code it is being used is better. Then a user would not need to jump to the struct definition to see what it is supposed to contain.

Reply:
IMO docs should be here in the header file.